src/ape_test/accounts.py

```diff
@@ -101,13 +101,17 @@ def alias(self) -> str:
     def address(self) -> AddressType:
         return self.network_manager.ethereum.decode_address(self.address_str)
 
-    def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
-        signed_msg = EthAccount.sign_message(msg, self.private_key)
-        return MessageSignature(
-            v=signed_msg.v,
-            r=to_bytes(signed_msg.r),
-            s=to_bytes(signed_msg.s),
-        )
+    def sign_message(self, msg: Any, **signer_options) -> Optional[MessageSignature]:
+        if isinstance(msg, str):
+            msg = encode_defunct(text=msg)
+        if isinstance(msg, SignableMessage):
```

Comment:
`elif` ?

Reply:
not the case here, I should add a comment to clarify

encode_defunct reutns a SignableMessage which we then want to process

the first str check can be thought of as pre-processing a string into a signable message before proceeding